includes/tax-functions.php

```diff
@@ -155,10 +155,19 @@ function edd_active_tax_rates_query_clauses( $clauses ) {
  *                          address information, then your store's Business Country setting.
  *                          Default true.
  *
- * @return mixed|void
+ * @return float
  */
 function edd_get_tax_rate( $country = '', $region = '', $fallback = true ) {
 
+	// This global is used to avoid excessive DB lookups per request.
+	global $edd_determined_tax_rates;
+
+	// First we only check for non-fallback options, because fallback logic happens later on.
+	$tax_rate_key = sanitize_key( strtolower( $country . '_' . $region ) );
```

Comment:
I'm reading through the logic of this function and it's confusing me completely on a few points:
- since all three parameters are optional, this `$tax_rate_key` _could_ end up being defined/used as `_`, which feels a little wonky. If the country and region are empty, should this bit be bypassed? Also I suppose you could end up with keys like `US_` if someone sets a coutry wide rate?
- I feel like line 172 is a little weird as we no longer use the option for the tax rate? At least we should not be?